### DISCUSSION TITLE
Is it possible to replace the "Search this site" search with a pnp-modern-search widget?

### DISCUSSION THREAD

**Discussion by catmanjan** [microsoft-search/pnp-modern-search#2525] [Q&A] [answered]
Sorry I am not sure if this is the right place to ask this, or if it is an existing feature...

At the top of all (?) SharePoint sites there is a searchbox which says "Search this site" you can type into it and it returns your results.

We already have a custom pnp-modern-search solution built, we are wondering if/how you can replace the "Search this site" feature with our own solution

Thanks

**Comment by wobba** [comment, marked as answer]
You can use something like this in the query template `{searchTerms} SiteID={Site.id}`. Also see https://microsoft-search.github.io/pnp-modern-search/usage/search-results/tokens/

**Reply by catmanjan** [reply]
Thanks that isn't quite what I'm looking for - I've done some more reading and worked out I need to use Set-PnPSearchSettings -Scope Web -SearchPageUrl https://mysearchpagehere

Is there a way to make this apply to all existing and new sites easily?

I basically want every search box at the top of the screen (even the one at my https://tenant.sharepoint.com/_layouts/15/sharepoint.aspx page) to go to a custom search page

Thanks

**Reply by wobba** [reply]
@catmanjan you have to script this, or have a custom site create solution which does it as a provisioning stage.
